dk-installer.py

```diff
@@ -1467,41 +1498,38 @@ class TestGenStartStep(Step):
 
     def execute(self, action, args):
         action.run_cmd(
-            "docker",
-            "compose",
-            "up",
-            "--wait",
-        )
+            get_docker_compose_base_command() + (
+                "up",
+                "--wait",
+            ))
 
     def on_action_fail(self, action, args):
         if action.args_cmd == "install":
-            action.run_cmd("docker", "compose", "down", "--volumes")
+            action.run_cmd(get_docker_compose_base_command() + ("down", "--volumes"))
 
 
 class TestGenStopStep(Step):
     label = "Stopping docker compose application"
 
     def execute(self, action, args):
         action.run_cmd(
-            "docker",
-            "compose",
+            get_docker_compose_base_command() + (
             "down",
-        )
+        ))
 
 
 class TestGenSetupDatabaseStep(Step):
     label = "Initializing the platform database"
 
     def execute(self, action, args):
         action.run_cmd(
-            "docker",
-            "compose",
+            get_docker_compose_base_command() + (
```

Comment:
Alternatively, your function can only deal with building the command parts, which I think it would be even better, especially if you end up using these commands also with the `start_cmd()`.

The call would look like this:

```python
 action.run_cmd(
    *tg_compose_cmd(
        "exec",
        "engine",
        "testgen",
        "setup-system-db",
         "--yes",
    )
)
```